# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gcc/gcc32/gcc-Os-fix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

This patch is extracted from
http://gcc.gnu.org/ml/gcc-patches/2004-01/msg00791.html

- Juergen "George" Sawinski <jsaw@gmx.net>

--- gcc-3.2.3/gcc/cfgcleanup.c.orig	2004-01-12 01:03:22.000000000 +0100
+++ gcc-3.2.3/gcc/cfgcleanup.c	2004-01-12 01:04:59.000000000 +0100
@@ -1640,7 +1640,7 @@
 		     /* If the jump insn has side effects,
 			we can't kill the edge.  */
 		     && (GET_CODE (b->end) != JUMP_INSN
-			 || (flow2_completed
+			 || (reload_completed
 			     ? simplejump_p (b->end)
 			     : onlyjump_p (b->end)))
 		     && merge_blocks (s, b, c, mode))
--- gcc-3.2.3/gcc/cfgrtl.c.orig	2004-01-12 01:03:40.000000000 +0100
+++ gcc-3.2.3/gcc/cfgrtl.c	2004-01-12 01:06:41.000000000 +0100
@@ -674,7 +674,7 @@
   if (tmp || !onlyjump_p (insn))
     return false;
 
-  if ((!optimize || flow2_completed) && tablejump_p (insn))
+  if ((!optimize || reload_completed) && tablejump_p (insn))
     return false;
 
   /* Avoid removing branch with side effects.  */
